ua/org.eclipse.tips.core/src/org/eclipse/tips/core/TipImage.java

```diff
@@ -205,15 +204,28 @@ public TipImage setAspectRatio(double aspectRatio) {
 
 	/**
 	 * Changes the default value "null" to the passed value which commonly is "png",
-	 * "gif" and such.
+	 * "gif" and updates the URL if the URL constructor was used or updates the
+	 * Base64Image if the base64 constructor was used.
 	 *
-	 * @param extension
-	 *            the extension of this file
+	 * @param newExtension the new extension of this file
 	 * @return this
 	 * @see #getExtension()
 	 */
-	public TipImage setExtension(String extension) {
-		fExtension = extension;
+	public TipImage setExtension(String newExtension) {
+		if (isURLSet()) {
+			String oldExtension = getExtension();
+			try {
+				fURL = new URL(fURL.toString().replace(oldExtension, newExtension));
+			} catch (MalformedURLException e) {
+				e.printStackTrace();
+			}
+		} else {
+			int from = fBase64Image.indexOf('/') + 1;
+			int to = fBase64Image.indexOf(';');
+			String oldExtension = fBase64Image.substring(from, to).trim();
+			fBase64Image = fBase64Image.replace(oldExtension, newExtension);
+		}
```

Comment:
Since there is no specification saying that changing the extension shall actually change the URL but only the tag in the image string, I'd propose to simply change this as follows:
```suggestion
				fBase64Image = fBase64Image.replaceAll("/.*;", "/" + extension + ";"); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
		}
```
Note that a replace operation based on the extension is not correct, as the substring with the extension can also appear at other places within the image string.